tools/pyboard.py

```diff
@@ -153,57 +153,114 @@ def inWaiting(self):
         else:
             return n_waiting
 
+# A thread that constantly reads from subp stdout and puts data into a queue (FIFO list)
+# When tested, it reads double '\r' which messes up the further code, so if double '\r' detected, skip one
+def read_stream(stream, q):
+    data_prec = b""
+    while True:
+        data = stream.read(1)
+        if (data_prec == data) and (data_prec == b'\r'):
+            continue
+        else:
+            q.put(data)
+        data_prec = data
 
 class ProcessToSerial:
```

Comment:
I think it'd be much cleaner to create separate classes, such as: `ProcessToSerialPosix` and `ProcessToSerialThreaded`.  Then some code like:
```python
import select
if hasattr(select, "poll"):
    ProcessToSerial = ProcessToSerialPosix
else:
    ProcessToSerial = ProcessToSerialThreading
```